### ISSUE TITLE
VST3 + Linux => No MIDI input?

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/Dplug#473]
> I've found that MIDI-in works great with the VST 2 version of Graillon 2.4, but is broken in the VST 3 version for me. I'm on Windows 10, Bitwig Studio 3.2.2.

**Comment by p0nce** [comment]
Seems broken in all Windows + VST3 situations (ABI?)

**Comment by p0nce** [comment]
Reproduced in Reaper + WIndows + 64-bit where MIDI input works in VST2.4 but not VST3

**Comment by p0nce** [comment]
What happens in practice:
```d
           int numEvents = eventList.getEventCount(); // this call works
            foreach(index; 0..numEvents)
            {
                Event e;
                if (eventList.getEvent(index, &e) == kResultOk) // after the call, event is default-initialized
                {
```

**Comment by p0nce** [comment]
Likely an ABI problem with the Event data structure

**Comment by p0nce** [comment]
Fixed in v9.2.10

**Comment by p0nce** [comment]
Fixed for mac 64-bit + VST3 in v9.2.11 (would not work in Studio One), probably brokend on Linux x86_64 too
(TODO: test on Linux + VST3)

**Comment by p0nce** [comment]
Confirmed bug on Linux too, and fixed in v9.2.15
